pkg/client/client.go

```diff
@@ -205,3 +214,44 @@ var addProviderVersionToUserAgent = request.NamedHandler{
 	Name: "openshift.io/cluster-api-provider-aws",
 	Fn:   request.MakeAddToUserAgentHandler("openshift.io cluster-api-provider-aws", version.Version.String()),
 }
+
+func resolveEndpoints(awsConfig *aws.Config, ctrlRuntimeClient client.Client, region string) error {
+	infra := &configv1.Infrastructure{}
+	infraName := client.ObjectKey{Name: globalInfrastuctureName}
+
+	if err := ctrlRuntimeClient.Get(context.Background(), infraName, infra); err != nil {
+		return err
+	}
+
+	// Do nothing when custom endpoints are missing
+	if infra.Spec.PlatformSpec.AWS == nil {
+		return nil
+	}
+
+	customEndpointsMap := buildCustomEndpointsMap(infra.Status.PlatformStatus.AWS.ServiceEndpoints)
```

Comment:
this would panic if either `PlatformStatus` or `PlatformStatus.AWS` are nil?

Reply:
we have a check on lines 227-229

Reply:
as far as I can see that's checking `infra.Spec.PlatformSpec.AWS`. That won't prevent this line from panicking.
I think the check above should include all the pointers that can be nil `infra.Status.PlatformStatus !=nil`, `infra.Status.PlatformStatus.AWS != nil`

Reply:
`PlatformSpec` is not a pointer here, it can't be nil

Reply:
PlatformStatus is a pointer, PlatformStatus.AWS is another pointer.
You are dereferencing them without checking they are nil.
am I missing anything?

Reply:
Sorry, you are right. I was looking at `PlatformSpec` :)  I pushed a fix for that.